make/lib/Awt2dLibraries.gmk

```diff
@@ -555,24 +555,40 @@ else
   ifeq ($(call isTargetOs, solaris), true)
     HARFBUZZ_CFLAGS += -DHAVE_SOLARIS_ATOMIC_OPS
   endif
-  # hb-ft.cc is not presently needed, and requires freetype 2.4.2 or later.
-  LIBFONTMANAGER_EXCLUDE_FILES += libharfbuzz/hb-ft.cc
+  
+   # Early re-canonizing has to be disabled to workaround an internal XlC compiler error
+   # when building libharfbuzz
+   ifeq ($(call isTargetOs, aix), true)
+     HARFBUZZ_CFLAGS += -qdebug=necan
+   endif
+
+   # hb-ft.cc is not presently needed, and requires freetype 2.4.2 or later.
+   # hb-subset and hb-style APIs are not needed, excluded to cut on compilation time.
+   LIBFONTMANAGER_EXCLUDE_FILES += hb-ft.cc hb-subset-cff-common.cc \
+       hb-subset-cff1.cc hb-subset-cff2.cc hb-subset-input.cc hb-subset-plan.cc \
+       hb-subset.cc hb-subset-instancer-solver.cc gsubgpos-context.cc hb-style.cc
 
    # list of disabled warnings and the compilers for which it was specifically added.
    # array-bounds         -> GCC 12 on Alpine Linux
    # parentheses          -> GCC 6
    # range-loop-analysis  -> clang on Xcode12
 
-  HARFBUZZ_DISABLED_WARNINGS_gcc := type-limits missing-field-initializers strict-aliasing \
-       array-bounds
-  # noexcept-type required for GCC 7 builds. Not required for GCC 8+.
-  # expansion-to-defined required for GCC 9 builds. Not required for GCC 10+.
-  HARFBUZZ_DISABLED_WARNINGS_CXX_gcc := reorder delete-non-virtual-dtor strict-overflow \
-       maybe-uninitialized class-memaccess unused-result extra noexcept-type expansion-to-defined
-  HARFBUZZ_DISABLED_WARNINGS_clang := unused-value incompatible-pointer-types \
-       tautological-constant-out-of-range-compare int-to-pointer-cast \
-       undef missing-field-initializers deprecated-declarations c++11-narrowing range-loop-analysis
-  HARFBUZZ_DISABLED_WARNINGS_microsoft := 4267 4244 4090 4146 4334 4819 4101 4068 4805 4138
+   HARFBUZZ_DISABLED_WARNINGS_gcc := type-limits missing-field-initializers strict-aliasing \
+        array-bounds parentheses dangling-pointer
```

Comment:
`parentheses dangling-pointer` are new here. `danling-pointer` seems to be part of the JDK 17 harfbuzz update to 10.4.0. `parenthesis` not. Is it needed?

Reply:
Not strictly required (gcc 14, gcc 4.8.5). Will remove, thanks!